### PR TITLE
Collision boxes and circles

### PR DESCRIPTION
**Pull request by stefankarschti**
Issues #1375 #1434 #1435 
Implementation of collision boxes and circles

### CODE REVIEW

**Comment by stefankarschti** [comment]
the static implementation does work. the shared one produces scrambled outputs, it may be that the buffers are changed after creation by the placement algorithm.

**Review by sjg-wdw — APPROVED** [review]
Looks pretty logical.
For the static vs shared issue since this is debugging output is this just a performance issue?

**Comment by TimSylvester** [comment]
I assume you checked on the raw types for the shared buffers?  I see `int2` vs `UByte2`, but that isn't necessarily a problem.

**Review by TimSylvester — APPROVED** [review]

**Comment by stefankarschti** [comment]
> Looks pretty logical.
> For the static vs shared issue since this is debugging output is this just a performance issue?

Yes, it is a performance issue. Memory gets copied if run in "static" mode.

**Comment by sjg-wdw** [comment]
For a debugging mode this doesn't worry me particularly.

**Comment by stefankarschti** [comment]
> I assume you checked on the raw types for the shared buffers? I see `int2` vs `UByte2`, but that isn't necessarily a problem.

Yes, I checked `int2` as well. It didn't make a difference. The `UByte2` came from `MBGL_DEFINE_ATTRIBUTE(uint8_t, 2, placed);` from [attribute.hpp](https://github.com/maplibre/maplibre-native/blob/6733dce7650a988962cac859f16c8a9595cf052a/src/mbgl/programs/attributes.hpp#L21C1-L21C43)

File: platform/ios/platform/ios/scripts/bazel-generate-plists.sh
Comment (mwilsnd) [inline platform/ios/platform/ios/scripts/bazel-generate-plists.sh:1]: This empty file should be safe to delete

**Review by mwilsnd — APPROVED** [review]

File: platform/ios/platform/ios/scripts/bazel-generate-plists.sh
Comment (stefankarschti) [reply platform/ios/platform/ios/scripts/bazel-generate-plists.sh:1]: on `topic/drawable-fixes` this is[ still not empty](https://github.com/maplibre/maplibre-native/blob/topic/drawable-fixes/platform/ios/platform/ios/scripts/bazel-generate-plists.sh) in use and referenced from
```
./platform/ios/platform/ios/scripts/bazel-xcodeproj.sh
./platform/ios/platform/ios/scripts/bazel-package.sh
./.github/workflows/ios-ci.yml
```

Comment (mwilsnd) [reply platform/ios/platform/ios/scripts/bazel-generate-plists.sh:1]: Right, I was just curious seeing it empty in this PR. Removing it here wouldn't have any effect on drawable-fixes when we merge it, but neither would just leaving it here as is.

Comment (stefankarschti) [reply platform/ios/platform/ios/scripts/bazel-generate-plists.sh:1]: I think I added an executable flag to it, since it was called from another script.

**Pull request merged**